src/providers/cryptoauthlib/key_operations.rs

```diff
@@ -0,0 +1,85 @@
+use super::key_slot::KeySlotStatus;
+use super::Provider;
+use crate::authenticators::ApplicationName;
+use log::{error, warn};
+use parsec_interface::operations::{psa_destroy_key, psa_generate_key};
+use parsec_interface::requests::{ResponseStatus, Result};
+
+impl Provider {
+    pub(super) fn psa_generate_key_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_generate_key::Operation,
+    ) -> Result<psa_generate_key::Result> {
+        let key_name = op.key_name;
+        let key_triple = self.key_info_store.get_key_triple(app_name, key_name);
+
+        self.key_info_store.does_not_exist(&key_triple)?;
```

Comment:
Talking about the other comment in the test file: this line should return `PsaErrorAlreadyExists` if the same key is created.